File: commands/dice.py
#!/usr/bin/env python3
import discord
from random import randint

async def dice(ctx):

    embed = discord.Embed(color=0xfee9b6,
                        title="⏳  Loading...")

    interaction = await ctx.respond(embed=embed)

    dice = randint(1,6)

    if dice == 1:
        roll = "You Rolled 1️⃣"
    if dice == 2:
        roll = "You Rolled 2️⃣"
    if dice == 3:
        roll = "You Rolled 3️⃣"
    if dice == 4:
        roll = "You Rolled 4️⃣"
    if dice == 5:
        roll = "You Rolled 5️⃣"
    if dice == 6:
        roll = "You Rolled 6️⃣" 

    embed = discord.Embed(color=0xea596e,
                          title="🎲   Dice Roll",
                          description=roll)

    await interaction.edit_original_response(embed=embed)

File: commands/dice.ts
import {
  DiscordInteraction,
  InteractionResponse,
  Colors,
} from "../types/discord";
import { embedResponse } from "../utils/discord";

const DICE_FACES = ["⚀", "⚁", "⚂", "⚃", "⚄", "⚅"];

export async function handleDice(
  _interaction: DiscordInteraction,
): Promise<InteractionResponse> {
  const roll = Math.floor(Math.random() * 6) + 1;
  const face = DICE_FACES[roll - 1];

  return embedResponse({
    title: "🎲 Dice Roll",
    description: `${face} You rolled a **${roll}**!`,
    color: Colors.PURPLE,
  });
}
